### PR TITLE
patch to mtsitrin/324: simplify block creation

### PR DESCRIPTION
**Pull request by danwt**
replaces https://github.com/dymensionxyz/dymint/pull/832, https://github.com/dymensionxyz/dymint/pull/830

### CODE REVIEW

File: block/submit.go [hunk 1]
```diff
@@ -126,22 +126,19 @@ func (m *Manager) HandleSubmissionTrigger() error {
 		return fmt.Errorf("create next batch to submit: %w", err)
 	}
 
-	if err := m.ValidateBatch(nextBatch); err != nil {
-		return fmt.Errorf("validate batch: %w", err)
-	}
-
 	resultSubmitToDA, err := m.submitNextBatchToDA(nextBatch)
```
Comment (omritoptix) [inline block/submit.go:129]: why not do here the same you did with the SL? ie calling the `m.DAClient.SubmitBatch` directly?

File: block/submit.go [hunk 2]
```diff
@@ -217,10 +187,15 @@ func (m *Manager) CreateNextBatchToSubmit(startHeight uint64, endHeightInclusive
 			// Remove the last block and commit from the batch
 			batch.Blocks = batch.Blocks[:len(batch.Blocks)-1]
 			batch.Commits = batch.Commits[:len(batch.Commits)-1]
+
+			if height == startHeight {
+				return nil, fmt.Errorf("batch size exceeds max size: %d", totalSize)
```
Comment (omritoptix) [inline block/submit.go:192]: in this case should probably say "block size exceeds max batch size" as its only one block.

File: block/submit.go (hunk 1 above)
Comment (danwt) [reply block/submit.go]: done

File: block/submit.go (hunk 2 above)
Comment (danwt) [reply block/submit.go]: done

**Review by omritoptix — APPROVED** [review]

**Pull request merged**